tests/Microsoft.Identity.Test.Unit.net45/ApiConfigTests/Harnesses/AcquireTokenInteractiveBuilderHarness.cs

```diff
@@ -0,0 +1,81 @@
+// 
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+// 
+// ------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client;
+using Microsoft.Identity.Client.ApiConfig;
+using Microsoft.Identity.Client.ApiConfig.Parameters;
+using Microsoft.Identity.Client.Utils;
+using Microsoft.Identity.Test.Common.Core.Helpers;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using NSubstitute;
+
+namespace Microsoft.Identity.Test.Unit.ApiConfigTests.Harnesses
+{
+    internal class AcquireTokenInteractiveBuilderHarness : AbstractBuilderHarness
+    {
+        public AcquireTokenInteractiveParameters InteractiveParametersReceived { get; private set; }
+        public IPublicClientApplication ClientApplication { get; private set; }
+
+        public async Task SetupAsync()
+        {
+            ClientApplication = Substitute.For<IPublicClientApplication, IPublicClientApplicationExecutor>();
+
+            await ((IPublicClientApplicationExecutor)ClientApplication).ExecuteAsync(
+                Arg.Do<AcquireTokenCommonParameters>(
+                    parameters =>
+                        CommonParametersReceived = parameters),
+                Arg.Do<AcquireTokenInteractiveParameters>(
+                    parameters =>
+                        InteractiveParametersReceived = parameters),
+                CancellationToken.None).ConfigureAwait(false);
+        }
+
+        public void ValidateInteractiveParameters(
+            IAccount expectedAccount = null,
+            IEnumerable<string> expectedExtraScopesToConsent = null,
+            string expectedLoginHint = null,
+            string expectedPromptValue = null,
+            bool expectedEmbeddedWebView = false)
+        {
+            Assert.IsNotNull(InteractiveParametersReceived);
+
+            Assert.AreEqual(expectedAccount, InteractiveParametersReceived.Account);
+            CoreAssert.AreScopesEqual(
+                (expectedExtraScopesToConsent ?? new List<string>()).AsSingleString(),
+                InteractiveParametersReceived.ExtraScopesToConsent.AsSingleString());
+            Assert.AreEqual(expectedLoginHint, InteractiveParametersReceived.LoginHint);
+
+            // todo(migration): shouldn't interactive request have a default prompt value?
```

Comment:
i think the default should be sending no prompt. i know for b2c, they don't want us to send a prompt and b2c will figure out which UI to show. if the customer wants something specific, they can set the parameter, but not sure if this will involve a larger cross-platform discussion. not sure what others are doing.

Reply:
right, but should the value of the Prompt string actually be null or should it be "no_prompt"?

---
In reply to: [249203077](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/800#discussion_r249203077) [](ancestors = 249203077)

Reply:
it should be null...sending prompt=none tells server to not show a prompt and we want to not sending anything for the prompt value. like we do now for b2c.

---
In reply to: [249210358](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/800#discussion_r249210358) [](ancestors = 249210358,249203077)